app/models/chat_channel.rb

```diff
@@ -65,66 +78,22 @@ def url
     "#{Discourse.base_url}/chat/channel/#{self.id}/-"
   end
 
-  def chatable_url
-    return nil if direct_message_channel?
-
-    chatable.url
-  end
-
-  def category_channel?
-    chatable_type == "Category"
-  end
-
-  def direct_message_channel?
-    chatable_type == "DirectMessageChannel"
-  end
-
-  def public_channel?
-    ChatChannel.public_channel_chatable_types.include?(chatable_type)
-  end
-
-  def chatable_has_custom_fields?
-    category_channel?
-  end
-
-  def allowed_user_ids
-    direct_message_channel? ? chatable.user_ids : nil
-  end
-
-  def allowed_group_ids
-    return if !category_channel?
-    return if category_channel? && !read_restricted?
-
-    staff_groups = Group::AUTO_GROUPS.slice(:staff, :moderators, :admins).values
-    chatable.secure_group_ids.to_a.concat(staff_groups)
-  end
-
   def public_channel_title
     chatable.name
   end
 
-  def read_restricted?
-    return true if direct_message_channel?
-    return chatable.read_restricted? if category_channel?
-
-    true
-  end
-
-  def title(user)
-    return chatable.chat_channel_title_for_user(self, user) if direct_message_channel?
-    return name if name.present?
-
-    title_from_chatable
+  def category_channel?
+    false
   end
+  alias direct_message_channel? category_channel?
+  alias public_channel? category_channel?
+  alias chatable_has_custom_fields? category_channel?
+  alias read_restricted? category_channel?
 
-  def title_from_chatable
-    case chatable_type
-    when "Category"
-      chatable.name
-    when "DirectMessageChannel"
-      chatable.chat_channel_title_for_user(self, user)
-    end
+  def allowed_user_ids
+    nil
   end
+  alias allowed_group_ids allowed_user_ids
```

Comment:
same should we mark one as deprecated?

Reply:
They are not used in the same contexts, I think one is expected for direct messages and the other one is expected for public channels. (so these methods are redefined in `DMChannel` and `CategoryChannel`)

Reply:
Ok sounds great, Im around all day if you need me to push this to the finish line.